### ISSUE TITLE
Spring 5 support for spring-kafka-test

### ISSUE THREAD

**Issue by fabienmifsud** [spring-projects/spring-kafka#719]
Hi all,

Can spring-kafka-test support spring 5 in the next release ?

I can help on that subject if needed.

Thx !

**Comment by artembilan** [comment]
Would you mind being more precise?
Spring Kafka project is based on Spring 5 since version `2.0`: https://docs.spring.io/spring-kafka/docs/2.0.0.RELEASE/reference/html/_introduction.html#compatibility

**Comment by fabienmifsud** [comment]
spring-kafka-test has spring-retry dependency wich refeers to a spring 4.x version and that leads my tests to have some class not found..

I did make a PR on it https://github.com/spring-projects/spring-retry/pull/117 to upgrade it.

**Comment by artembilan** [comment]
Well, that's not so easy to move Spring Retry to Spring 5, because the last one requires Java 8 minimum and for that reason we would need to start a version `2.0`.
That even might be a bad idea, but we don't have themes for `2.0` version yet.
On the other hand we haven't done that yet just because the current Spring Retry version is fully compatible with Spring 5 - we just only need to have proper dependencies in the classpath!

To fix the problem in your project you need to `exclude` Spring 4 transitive dependencies from the spring-retry dependency:

```
compile ("org.springframework.retry:spring-retry:1.2.2.RELEASE") {
    exclude group: 'org.springframework'
}
```

I would say that exactly this fix has to be done here in this Spring Kafka project.

**Comment by fabienmifsud** [comment]
I'm not sure to well understand why you don't want to upgrade versions, but
ok. I did try to exclude spring-retry and spring-core from transitive test
dependencies and it hasn't been enough. I'll try to exclude the whole
groupId and let you know. Thanks for your time

Le ven. 29 juin 2018 à 17:07, Artem Bilan <notifications@github.com> a
écrit :

**Comment by artembilan** [comment]
Just bumping version is not enough to start a new major version.
We can't do what you suggest in your PR in the current minor version: it is going to be a big breaking change for our users and we are very reluctant to make breaking change.
Another argument do not start a new major version I've explained before: Spring Retry works well with the latest Spring 5.0 - just need to adjust dependencies in your project a bit!
Otherwise we would need to support two major versions in parallel. 

Nevertheless, there is no reason to start a new major version because everything is compatible in the current state.

I didn't tell you to exclude `spring-retry`. Its Spring transitive dependencies must be excluded.
We do that explicitly in other Spring 5 based projects, but have missed somehow here in Spring Kafka. That's what I suggest to do here in borders of this ticket.

Your PR to Spring Retry is a different story and I already explained my vision for that: I'm afraid it is going to be reject as non-valuable and fully breaking changes fix.
